### ISSUE TITLE
Auto-Renable Secret Service Integration when having been enabled but had another service running

### ISSUE THREAD

**Issue by hasezoey** [keepassxreboot/keepassxc#8558]
## Summary
[NOTE]:
Currently the SSI in KeepassXC will *stay* disabled when another service had been running, and does not re-enable itself after that service is gone

## Examples
[NOTE]:
type 1:
Instead of the warning being shown that there is already a keyring process running and removing the checkbox from `Enable KeepassXC Secret Service Integration`, it should stay enabled and auto-start once no other process is detected
type 2:
same as type 1 but behind a extra option

## Context
[NOTE]:
- This feature matters for me because sometimes the gnome-keyring gets started before keepassxc and needs to be stopped, but after that process has been stopped, keepassxc does not automatically re-enable SSI when it had been enabled before (which can lead to the second in this list because of me forgetting to re-enable it).
- This also matters because if for some reason keepassxc did not register SSI and i start something like vscode (a application that requests something from a keyring), it will auto-start gnome-keyring and needs to be manually stopped (like the case above) and in keepassxc manually re-enabled.

(i also had already considered removing gnome-keyring, but it seems tied to `xdg-desktop-portal-gnome`, which is required by other packages)
Manjaro Linux using KDE Plasma Wayland

**Comment by michaelk83** [comment]
You can try https://github.com/keepassxreboot/keepassxc/issues/6274#issuecomment-810983553 as a workaround. This should take care of auto-start via DBus, when needed by a client (the 2nd point in your context). For the 1st point, you'll also need to check for other ways gnome keyring may auto-start during login, such as PAM and XDG auto-start, and disable those. It might have a setting for that ("Start at login" or something), not sure.

**Comment by hasezoey** [comment]
> You can try https://github.com/keepassxreboot/keepassxc/issues/6274#issuecomment-810983553 as a workaround. This should take care of auto-start via DBus, when needed by a client (the 2nd point in your context)

thanks, will try that

> For the 1st point, you'll also need to check for other ways gnome keyring may auto-start during login, such as PAM and XDG auto-start, and disable those

well i have it disabled (via xdg autostart `Hidden=true`), but some way or another it sometimes just pops up again (maybe because of the dbus thing)

though i would still like a setting that automatically re-enables the KeepassXC SSI when no other process for that is running

**Comment by droidmonkey** [comment]
> (i also had already considered removing gnome-keyring, but it seems tied to xdg-desktop-portal-gnome, which is required by other packages)

I hate that gnome-keyring is like the internet explorer of Linux. It's not even a dependant software because it communicates over dbus.

I feel like a bug report needs to be filed against Manharo package management to remove that dependency.

**Comment by hasezoey** [comment]
> I feel like a bug report needs to be filed against Manharo package management to remove that dependency.

i just noticed that `xdg-desktop-portal-gnome` is actually not required and other things (optionally) depend on `org.freedesktop.secrets` which keepassxc (the package at least) provides, so i could safely remove it without breaking things

---

i will keep this open as a feature request anyway (though original "why does it matter (for me)" is now resolved)

**Comment by droidmonkey** [comment]
I agree this feature would be desirable. Perhaps behind a secondary checkbox like "Automatically restart service when possible".
